### ISSUE TITLE
Emulate (CONTROL T)

### ISSUE THREAD

**Issue by pdewacht** [jeffshrager/elizagen.org#9]
This depends on how much historical accuracy we really want... The original code uses (CONTROL T), which turns off line buffering. (See BBN Lisp documentation at: http://bitsavers.trailing-edge.com/pdf/bbn/The_BBN-LISP_System_Apr69.pdf)

This probably means that the doctor is intended to answer the patient immediately when they finish a sentence, without them needing to press the Return key. The start-up instructions also hint at this:

```
PLEASE TERMINATE INPUT WITH A PERIOD OR A QUESTION MARK.
```

Not implementing this behavior means that patients get confusing output when they type two sentences on a single line. And we probably also print a spurious empty line before the doctor's responses.

There seems no portable way to implement this. On Unix you'd need to fiddle with termios, and on Windows there's the Console API. There doesn't seem to be any Lisp library that provides a portable interface to this functionality.

**Comment by ambiamber** [comment]
@pdewacht

Thanks for pointing out that an SDS 940 has a status key!
I've been collecting status keys for quite a while and I've never seen any old big iron other than from DEC with a status key.
https://en.wikipedia.org/wiki/Status_key

I added the SDS 940 to the Talk page of the Status Key article for now.  Once I understand it enough I will add it to the article. I hope you actually read the Status Key page because you have misconceptions about CONTROL-T and it's long history. You wrote:

> turns off line buffering

That is entirely incorrect.

**Comment by larsbrinkhoff** [comment]
@ambiamber, agreed, very interesting!  TENEX was influenced by the Berkeley system, so TOPS-20 must have got it from there.

I didn't see the information in this issue, but I found it on page 23.10 in the linked pdf.  Added here for completeness:
![control-t](https://user-images.githubusercontent.com/775050/208235368-946c7406-8f96-4af0-91ba-771e685dae0d.png)

**Comment by pdewacht** [comment]
@ambiamber You misunderstood me, I wasn't talking about a Ctrl-T keystroke.

The Eliza code has the function call `(control t)` near the top: https://github.com/jeffshrager/elizagen.org/blob/5671ce249fae09c34eef412169c8383662948a0b/1966_Cosell_BBNLISP/coselleliza1969and1972.transcript#L61

The `control` function is documented on page 14.10 of the BBN Lisp manual:
![afbeelding](https://user-images.githubusercontent.com/223495/208241796-174941b7-6751-45f0-af95-0c354f4e23c0.png)

So the effect of the `(control t)` call is to turn off line buffering.

**Comment by ambiamber** [comment]
@pdewacht [never mind](https://www.youtube.com/watch?v=HE6EP3Z0PSU).

**Comment by ambiamber** [comment]
> TOPS-20 must have got it from there

DEC RSTS/E has a status key and it came before TOPS-20 too. (post hoc ergo propter hoc).

**Comment by larsbrinkhoff** [comment]
It's not a question about which came before the other.

**Comment by ambiamber** [comment]
Intent is the most difficult thing to prove.
